controllers/macvlannetwork_controller.go

```diff
@@ -89,10 +93,23 @@ func (r *MacvlanNetworkReconciler) updateCrStatus(cr *mellanoxcomv1alpha1.Macvla
 		cr.Status.Reason = syncError.Error()
 	}
 
+	netAttachDef := &netattdefv1.NetworkAttachmentDefinition{}
```

Comment:
I wonder if we should conditionally do this bit only if cr.Status is "Ready", else set the net-attach-def self link to empty.

This will save some un-needed API Calls.